src/extract.jl

```diff
@@ -154,12 +154,15 @@ function git_file_hash(
     # TODO: this largely duplicates the logic of read_data
     # read_data could be used directly if SHA offered an interface
     # where you write data to an IO object and it maintains a hash
+    t = round_up(size)
     while size > 0
-        r = readbytes!(tar, buf, size < sizeof(buf) ? round_up(size) : sizeof(buf))
-        r < 512 && eof(io) && error("premature end of tar file")
+        n = min(t, length(buf))
+        r = readbytes!(tar, buf, n)
```

Comment:
This is wasteful if you're not at the end of the tar file

Reply:
I'd just keep the logic as is and check `r % 512 != 0`

Reply:
Why? It reads one full buffer’s worth at a time until it’s read the full amount. What’s the inefficiency?

Reply:
I think I misread the code. This seems fine.